src/AppInstallerCLICore/FontInstaller.cpp

```diff
@@ -57,8 +57,24 @@ namespace AppInstaller::CLI::Font
                 title += s_TrueType;
             }
 
-            AICLI_LOG(CLI, Info, << "Moving font file to: " << destinationPath);
-            AppInstaller::Filesystem::RenameFile(filePath, destinationPath);
+            // If font subkey already exists, remove the font file if it exists.
+            if (m_key[title].has_value())
+            {
+                AICLI_LOG(CLI, Info, << "Existing font subkey found:" << AppInstaller::Utility::ConvertToUTF8(title));
+                std::filesystem::path existingFontFilePath = { m_key[title]->GetValue<Registry::Value::Type::String>() };
+
+                if (m_scope == Manifest::ScopeEnum::Machine)
+                {
+                    // Font entries in the HKEY_LOCAL_MACHINE hive only have the filename specified as the value. Prepend install location.
+                    existingFontFilePath = m_installLocation / existingFontFilePath;
+                }
+
+                if (std::filesystem::exists(existingFontFilePath))
+                {
+                    AICLI_LOG(CLI, Info, << "Removing existing font file at:" << existingFontFilePath);
+                    std::filesystem::remove(existingFontFilePath);
```

Comment:
If the operation fails, we cannot revert this.  Much like my comment on the rest of the operation being reverted on failure, we want some mechanism to put things back the way they were.

I would think that to make it maximally recoverable, one would need to:
1. Create a temporary portable tracking database for the file and registry changes (probably requires some amount of addition to the tracked things)
2. Set up a resume that can leverage the database to revert to the previous state

But I don't think we need to be that extreme (handling full on process termination).  A more modest approach would have a vector of `IOperationSteps` that you populate as you go.  On exiting the operation, you either invoke `Complete` or `Revert` on all of the items, depending on success or failure.  In this case, `Complete` deletes the renamed file, while `Revert` renames it back to its old state.